src/components/components.cpp

```
@@ -46,107 +46,104 @@ namespace algorithms {
-        throw;
-    }
-
-    // get the results
-    std::vector< std::vector< int64_t > > results(num_comps);
-    for (auto vd : boost::make_iterator_range(vertices(graph.graph))) {
-        results[components[vd]].push_back(graph[vd].id);
-    }
-
-    return detail::componentsResult(results);
+  typedef pgrouting::UndirectedGraph::V V;
+  // perform the algorithm
```

Cppcheck (2.20.0)

[information] Include file

(missingIncludeSystem)